tests/test_utils.py

```diff
@@ -281,6 +290,19 @@ def test_parse_query_string(self):
         assert result['f'] == ['a', 'a=b']
         assert result['é'] == 'a=b'
 
+    @pytest.mark.parametrize('query_string,keep_blank,expected', [
+        ('', True, {}),
+        ('', False, {}),
+        ('flag1&&&&&flag2&&&', True, {'flag1': '', 'flag2': ''}),
+        ('flag1&&&&&flag2&&&', False, {}),
+        ('malformed=%FG%1%Hi%%%a', False, {'malformed': '%FG%1%Hi%%%a'}),
+        ('spade=♠&spade=♠', False, {'spade': ['♠', '♠']}),  # Unicode query
+    ])
```

Comment:
Could you also test cases with strings that contain 2+ `=` characters in a row? Also, test strings that contain a field that starts with an `=` character (key is missing), either at the beginning of the string or immediately following an `&`. Besides these, feel free to dream up any other strange inputs a vulnerability scanner might throw at us.

Reply:
Will do.
Indeed, it makes sense to have a very solid unit test selection here, due to the gravity of operating on raw pointers. If there is some shady edge case in Python code for one request in a million, no one is probably going to care, however getting a `SIGSEGV` in a web framework worker process is unlikely to go unnoticed :slightly_smiling_face: